beacon-chain/db/kv/blocks.go

```diff
@@ -800,11 +807,11 @@ func unmarshalBlock(_ context.Context, enc []byte) (interfaces.SignedBeaconBlock
 }
 
 // marshal versioned beacon block from struct type down to bytes.
-func marshalBlock(_ context.Context, blk interfaces.SignedBeaconBlock) ([]byte, error) {
+func marshalBlock(_ context.Context, blk interfaces.SignedBeaconBlock, saveBlindedBlocks bool) ([]byte, error) {
 	var encodedBlock []byte
 	var err error
 	blockToSave := blk
-	if features.Get().EnableOnlyBlindedBeaconBlocks {
+	if saveBlindedBlocks {
```

Comment:
I set this if branch to always be true and no tests failed. Are you sure you have enough test coverage on marshalBlock with this new functionality?

Reply:
After refactoring, confirmed we now have have coverage in both branches. It has been split into two functions and we now have them covered